### ISSUE TITLE
Getting "use of undeclared type or module"

### ISSUE THREAD

**Issue by davidpdrsn** [dtolnay/trybuild#1]
I'm getting this error

```
Testing tests/compile_pass/simple_non_null_scalars.rs ... error
----------------
error[E0432]: unresolved import `juniper`
 --> $DIR/simple_non_null_scalars.rs:3:5
  |
3 | use juniper::{EmptyMutation, Executor, FieldResult, Variables};
  |     ^^^^^^^ use of undeclared type or module `juniper`
```

but `juniper = "0.11.0"` is in my "dependencies". Also tried adding it to "dev_dependencies" but that didn't help.

**Comment by dtolnay** [comment]
Hi! I am trying to reproduce the behavior you are seeing. Here is what I tried:

```console
$ cat Cargo.toml 
[package]
name = "davidpdrsn"
version = "0.0.0"
edition = "2018"

[dev-dependencies]
juniper = "0.11"
trybuild = "1.0"
```

```console
$ cat tests/ui.rs 
#[test]
fn ui() {
    let t = trybuild::TestCases::new();
    t.pass("tests/compile_pass/simple_non_null_scalars.rs");
}
```

```console
$ cat tests/compile_pass/simple_non_null_scalars.rs 
use juniper::{EmptyMutation, Executor, FieldResult, Variables};

fn main() {
    let _: EmptyMutation<()>;
    let _: Executor<()>;
    let _: FieldResult<()>;
    let _: Variables;
}
```

But with this setup, `cargo test` passes. Does this minimal setup pass on your computer or does it fail the same way as your real one? Can you tell what relevant differences there might be between the two?

I have been undecided about whether to exposes [dependencies] in addition to [dev-dependencies], so for now it was only providing [dev-dependencies] to the test case. I found that `cargo test` natively makes both available, so I went ahead and mirrored that behavior in trybuild 1.0.3. I would still like to figure out what was going wrong even if the same code now works with 1.0.3.

**Comment by davidpdrsn** [comment]
1.0.3 seems to have fixed it. Thanks!

**Comment by dtolnay** [comment]
I figured it out -- https://github.com/davidpdrsn/juniper-from-schema/blob/0.2.0/Cargo.toml#L31 uses `[dev_dependencies]` instead of the `[dev-dependencies]` [shown in Cargo documentation](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#development-dependencies). I guess Cargo accepts both. I released a fix in trybuild 1.0.4 so you should be able to move juniper back to dev-dependencies.
